jbpm-bpmn2/src/main/java/org/jbpm/bpmn2/xml/IntermediateThrowEventHandler.java

```diff
@@ -241,9 +241,15 @@ public void handleMessageNode(final Node node, final Element element,
 							"Could not find message " + messageRef);
 				}
                 message.addOutgoingNode(node);
+                String varName = null;
                 String mappingVariable = (String) actionNode.getMetaData("MappingVariable");
                 Variable v = (Variable) ((ProcessBuildData) parser.getData()).getMetaData("Variable");
-                String varName = v != null ? (String) v.getMetaData(mappingVariable) : mappingVariable;
+                if (v != null) {
```

Comment:
In previous implementation if Variable was not null but Variable.getMedata(mappingVariable) was, then rather that use mappingVariable (as I think should be the case) we will be setting varName to null. 
This makes the message object used by the action (which is also used for Kafka  extension) null.